proposals/pr-contrib-guide.md

```diff
@@ -0,0 +1,315 @@
+determination include:
+
+1. The change will be made on part of the code base that the maintainers have
+   determined to be available. See the FAQ for more details.
+1. There is evidence that the change will be impactful for a significant number
+   of users beyond the individual that proposed it.
+1. The background, description, and use cases for the feature are clearly
+   defined.
+1. There are test cases and edge cases defined.
+1. There is consensus with the maintainers on an implementation plan.
+
+Once these criteria have been met, an issue is available for the community to
+contribute. If it's not something that the maintainers feel should be added, the
+issue should be left for community discussion and upvotes. A comment should be
+added by the maintainers to describe how users can advocate for the feature with
+more use cases, or vote for the issue using reactions. The issue is subject to
+existing stale and ancient issue closure mechanisms.
+
+When an issue is being worked on by the community, maintainers will provide
+guidance and support during the contribution process. They will engage with the
+community during discussion of implementation details or issues that arise
+during the contribution. Notably, we will initially respond within [TBD]
+business days on a PR review request. This does not guarantee how long it will
+be until the PR is merged.
+
+When a pull request review is made, the maintainers will review it and make a
+decision if it can be included or more work is needed. For a pull request to be
+complete, it needs:
+
+1. Have an open issue detailing an implementation agreed upon by the
+   maintainers.
+1. Implement the change in the way described in the issue.
+1. Pass all validation checks, including tests and linting.
+1. Have an approved review from a maintainer.
+
+Once a pull request is complete with to the satisfaction of the maintainers, it
+can be merged and acknowledged as contribution from the community.
+
+We do not want to discourage users to contribute or make proposed changes, but
+we need to acknowledge that the maintainers still have limited bandwidth to give
+feedback on implementation or design and review pull requests. If an issue has
+not been explicitly marked as being available for contribution, it should not be
+expected that a pull request will get a review. We should clearly indicate which
+user contributions the maintainers are actively engaged on.
+
+If a pull request becomes stale because the contributor is no longer actively
+engaged or working on it for more than [TBD] days, it should be reviewed to see if
+it is suitable for the maintainers to finish the work or marked as stale and
+documented for another user to continue working on it.
+
+![Figure A: Flowchart of contribution guide review
+process.](contrib-guide-high-level.png "Figure A: Flowchart of contribution
+guide review process.")
+
+**Figure A: Flowchart of contribution guide review process.**
```

Comment:
It would be good to hoist this diagram to the top of the specification, as it is fairly hard to follow what the proposed changes are without it.